processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -157,6 +163,45 @@ public void remove()
     );
   }
 
+  public static ValueMatcher getTimeValueMatcher(
+      final LongColumnSelector longSelector,
+      Comparable value
+  )
+  {
+    try {
+      if (value == null) {
+        return new BooleanValueMatcher(false);
+      }
+      final long longValue = Long.parseLong(value.toString());
+      return new ValueMatcher()
+      {
+        @Override
+        public boolean matches()
+        {
+          return longSelector.get() == longValue;
+        }
+      };
+    }
+    catch (NumberFormatException nfe) {
+      throw new ParseException("Cannot parse value [%s] as long: ", value);
+    }
+  }
+
+  public static ValueMatcher getTimePredicateMatcher(
+      final LongColumnSelector longSelector,
+      final Predicate predicate
```

Comment:
`Predicate<Object>` expresses intent a bit better. This should also have some javadoc explaining how the predicate should work.